plugins/outputs/elasticsearch/elasticsearch.go

```diff
@@ -475,6 +495,32 @@ func (a *Elasticsearch) GetIndexName(indexName string, eventTime time.Time, tagK
 	return fmt.Sprintf(indexName, tagValues...)
 }
 
+func (a *Elasticsearch) GetPipelineName(pipelineInput string, tagKeys []string, metricTags map[string]string) string {
+	if !strings.Contains(pipelineInput, "%") {
+		return pipelineInput
+	}
+
+	var (
+		tagValues          []interface{}
+		useDefaultPipeline bool
+	)
+
+	for _, key := range tagKeys {
+		if value, ok := metricTags[key]; ok {
+			tagValues = append(tagValues, value)
+		} else {
+			a.Log.Debugf("Tag %s not found, reverting to default pipeline instead.", key)
+			useDefaultPipeline = true
+		}
+	}
+
+	if len(tagValues) > 0 && !useDefaultPipeline {
+		return fmt.Sprintf(pipelineInput, tagValues...)
+	}
+
+	return a.DefaultPipeline
```

Comment:
Can we write this shorter as
```suggestion
	if len(tagKeys) == 0 {
		return ""
	}
	
	for _, key := range tagKeys {
		value, ok := metricTags[key]
		if !ok {
			a.Log.Debugf("Tag %s not found, reverting to default pipeline instead.", key)
			return a.DefaultPipeline
		}
		
		tagValues = append(tagValues, value)
	}

	return fmt.Sprintf(pipelineInput, tagValues...)
```

Reply:
I was looking at taking this approach to the function, however this would not return the default pipeline as a fallback.. and if one of the tag values is missing it will return an improper string in its place.. With the first version it will log the missing tag key for debugging and also fall back to the default pipeline to be processed.

Reply:
Ah I see. However, if you change line 510 into `return a.DefaultPipeline` the two are equivalent. Please note the `return` statement in the for-loop. So as soon as one tag is _not_ found, the default-pipeline is returned.